### PR TITLE
Scram SHA 256 

### PR DESCRIPTION
**Pull request by janciesla8818**
This PR adds SCRAM SHA 256 as SASL mechanism. This is the first iteration of the functionality. So far it works when passwords are stored as paintext.

Changes include:
* Pointing to current escalus that is supporting sha-256.
* Modification of login tests for testing with sha-256 login.
* Adding cyrsasl_scram_sha256 and mongoose_scram_sha256 modules. The  modules are copies of cyrsasl_scram and mongoose_scram modules. Only difference is that they are using sha256 for hashing. 
* Updated documentation to reflect the changes.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[8050.1](https://travis-ci.org/esl/MongooseIM/jobs/666432696) / Erlang 22.0 / small_tests / 17e5068b4f45d8cd6a51adc78b6bbbfdde7269e5
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2685/8050/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2685/8050/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-03-24_17.07.22/lib.mongooseim.logs/run.2020-03-24_17.07.22/suite.log.html)

File: src/sasl/cyrsasl_scram_sha256.erl [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+%%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+%%% General Public License for more details.
+%%%
+%%% You should have received a copy of the GNU General Public License
+%%% along with this program; if not, write to the Free Software
+%%% Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+%%%
+%%%----------------------------------------------------------------------
+
```
Comment (michalwski) [inline src/sasl/cyrsasl_scram_sha256.erl]: To be honest, I think we don't need this section. First of all, the module is new, created by us. Second of all the license is in the root directory, so (IMO) there is no need to repeat it in every module (unless I'm mistaken).
```suggestion
```

File: src/sasl/cyrsasl_scram_sha256.erl [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+-module(cyrsasl_scram_sha256).
+
+-author('stephen.roettger@googlemail.com').
+
```
Comment (michalwski) [inline src/sasl/cyrsasl_scram_sha256.erl]:
```suggestion
```

File: src/mongoose_scram_sha256.erl [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+%%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+%%% General Public License for more details.
+%%%
+%%% You should have received a copy of the GNU General Public License
+%%% along with this program; if not, write to the Free Software
+%%% Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+%%%
+%%%----------------------------------------------------------------------
+
```
Comment (michalwski) [inline src/mongoose_scram_sha256.erl]: We really don't need this.
```suggestion
```

File: src/mongoose_scram_sha256.erl [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+-module(mongoose_scram_sha256).
+
+-author('stephen.roettger@googlemail.com').
+
```
Comment (michalwski) [inline src/mongoose_scram_sha256.erl]:
```suggestion
```

**Comment by mongoose-im** [comment]
[8052.1](https://travis-ci.org/esl/MongooseIM/jobs/666718186) / Erlang 22.0 / small_tests / fed0207876b34814f188557e1a0b26cbe2bdb7d9
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2685/8052/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2685/8052/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-03-25_10.32.10/lib.mongooseim.logs/run.2020-03-25_10.32.10/suite.log.html)

**Pull request merged**